server/lib/src/filter.rs

```diff
@@ -1504,6 +1526,7 @@ impl FilterResolved {
                 FilterResolved::resolve_no_idx((*f).clone(), ev)
                     .map(|fi| FilterResolved::AndNot(Box::new(fi), None))
             }
+            FilterComp::Invalid(_) => None,
```

Comment:
Needs to return Some(FilterResolved::Invalid ) else this causes the filter element to be dropped from the query.

Reply:
See line 539 in filter.rs.

Reply:
Wouldn't that be desirable? An invalid filter shouldn't make it through and be dropped saving us a few CPU cycles. Or do we want to keep the query perfectly in tact to preserve "correctness"

Reply:
This component may be invalid, but the rest is valid. Consider the original reason for this issue - `|( (uid=?)(spn=?) )`

spn would be invalid, and then with your suggestion, the whole question would instant fail because the spn couldn't "resolve" the uuid. 

When the real answer is that spn will never match (invalid) but uid= can match, and then causes the whole OR condition to be true.

Reply:
Strangely the observed behavior was that it worked correctly in any case no matter if I returned None or Some. So I am unsure to what degree this is actually applicable but I changed the code just to be safe.